mobile/src/main/java/org/openhab/habdroid/core/message/MessageHandler.java

```diff
@@ -0,0 +1,128 @@
+package org.openhab.habdroid.core.message;
+
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+import android.support.annotation.Nullable;
+import android.support.design.widget.Snackbar;
+import android.util.Log;
+import android.view.View;
+import android.widget.Toast;
+
+import org.openhab.habdroid.util.Constants;
+
+public class MessageHandler {
+    private static final String TAG = MessageHandler.class.getSimpleName();
+    public static final int TYPE_DIALOG = 1;
+    public static final int TYPE_SNACKBAR = 2;
+    public static final int TYPE_TOAST = 3;
+
+    public static final int LOGLEVEL_DEBUG = 0;
+    public static final int LOGLEVEL_REMOTE = 1;
+    public static final int LOGLEVEL_LOCAL = 2;
+    public static final int LOGLEVEL_NO_DEBUG = 4;
+    public static final int LOGLEVEL_ALWAYS = 5;
+
+    private static Snackbar snackbar;
+    private static AlertDialog alertDialog;
+    private static Toast toast;
```

Comment:
Storing UI elements in static variables can leak lots of UI state permanently, thus IMHO must be avoided. Better make MessageHandler a member variable in the activity, passing the activity into MessageHandler's constructor.

I'll add a commit doing that in my branch for your cherry picking convenience.